types/memoize-one/index.d.ts

```diff
@@ -3,7 +3,9 @@
 // Definitions by: Karol Majewski <https://github.com/karol-majewski>, Frank Li <https://github.com/franklixuefei>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-declare function memoizeOne<T extends (...args: any[]) => any>(resultFn: T, isEqual?: EqualityFn): T;
-export type EqualityFn = (a: any, b: any, index: number) => boolean;
+declare function memoizeOne<T extends (...args: any[]) => any>(resultFn: T, isEqual?: memoizeOne.EqualityFn): T;
+declare namespace memoizeOne {
+    type EqualityFn = (a: any, b: any, index: number) => boolean;
+}
 
-export default memoizeOne;
+export = memoizeOne;
```

Comment:
`memoize-one` is distributed in a few formats: CommonJS, ESM, and UMD.

```
memoize-one.cjs.js
memoize-one.cjs.js.flow
memoize-one.esm.js
memoize-one.js
memoize-one.min.js
```

It then defines two entry points in [`package.json`](https://github.com/alexreardon/memoize-one/blob/d143f883ff81dba9c46f148d212337667b941a99/package.json):

-  "main": "dist/memoize-one.cjs.js",
-  "module": "dist/memoize-one.esm.js",

Since `memoize-one.esm.js` does include a `default export`, but it's not distributed under the `main` field, does it mean the change in this PR was justified and we should treat it as if it did not have a default export?

Reply:
It's justified. TS doesn't use the `module` field, so only `main` matters.

Reply:
It's not just TS, it's node. We're trying to match the lowest-common-denominator usage:

```
$ node
> require('memoize-one').default
```

If that's undefined, then the module doesn't have a default export.

Other loaders may behave differently, but not everyone is guaranteed to be using those loaders. If they are, then they can turn on `esModuleInterop`.